client/src/main/com/sinch/sdk/domains/verification/adapters/StatusService.java

```diff
@@ -14,22 +14,29 @@
 import com.sinch.sdk.domains.verification.models.VerificationReport;
 import com.sinch.sdk.models.Configuration;
 import java.util.Map;
+import java.util.function.Supplier;
 
 public class StatusService implements com.sinch.sdk.domains.verification.StatusService {
 
-  private QueryVerificationsApi api;
-
-  public StatusService() {}
+  private final Configuration configuration;
+  private final HttpClient httpClient;
+  private final Supplier<Map<String, AuthManager>> authManagerSupplier;
 
   public StatusService(
-      Configuration configuration, HttpClient httpClient, Map<String, AuthManager> authManager) {
-    this.api =
-        new QueryVerificationsApi(
-            httpClient, configuration.getVerificationServer(), authManager, new HttpMapper());
+      Configuration configuration,
+      HttpClient httpClient,
+      Supplier<Map<String, AuthManager>> authManagerSupplier) {
+    this.configuration = configuration;
+    this.httpClient = httpClient;
+    this.authManagerSupplier = authManagerSupplier;
   }
 
-  private QueryVerificationsApi getApi() {
-    return this.api;
+  protected QueryVerificationsApi getApi() {
+    return new QueryVerificationsApi(
```

Comment:
Exact, now the credentials are dynamic than static; but this far to be perfect in context of Immutable pattern (see current ADR comments and previous comment because this cannot be the final implementation)

Reply:
All right, let's keep it this way for the moment then